include/kodlab_mjbots_sdk/pid.h

```diff
@@ -0,0 +1,192 @@
+/**
+ * @file pid.h
+ * @author Chandravaran Kunjeti (kunjeti@seas.upenn.edu)
+ * @brief Class implementation of PID
+ * @date 9/30/2022
+ *
+ * @copyright 2022 The Trustees of the University of Pennsylvania. All rights reserved.
+ *
+ */
+
+#pragma once
+
+#ifndef _PID_
+#define _PID_
+
+#include "Eigen/Geometry"
+
+namespace kodlab
+{
+
+  /**
+   * @brief Pid object
+   */
+  class Pid
+  {
+
+  public:
+    /**
+     * @brief Construct a Pid object
+     * @param p_gain Proportional Gain
+     * @param d_gain Derivative Gain
+     * @param i_gain Integral Gain
+     * @param time_step Time difference between 2 updates
+     */
+    Pid(double p_gain, double d_gain, double i_gain, double time_step, double deadband, double saturation)
+        : kp_(0), kd_(0), ki_(0), time_step_(0.01), deadband_(0), saturation_(0)
+    {
+
+      kp_ = p_gain;
+      kd_ = d_gain;
+      ki_ = i_gain;
+      time_step_ = time_step;
+      deadband_ = deadband;
+      saturation_ = saturation;
+    }
+
+    /**
+     * @brief Function to set the PID gains
+     * @param p_gain Proportional Gain
+     * @param d_gain Derivative Gain
+     * @param i_gain Integral Gain
+     */
+    void SetGains(double p_gain, double d_gain, double i_gain)
+    {
+
+      kp_ = p_gain;
+      kd_ = d_gain;
+      ki_ = i_gain;
+    }
+
+    /**
+     * @brief Function to set the deadband
+     * @param deadband Deadband that needs to be set
+     */
+    void SetDeadband(double deadband)
+    {
+
+      deadband_ = deadband;
+    }
+
+    /**
+     * @brief Function to set the Saturation
+     * @param saturation Saturation
+     */
+    void SetSaturation(double saturation)
+    {
+
+      saturation_ = saturation;
+    }
+
+    /**
+     * @brief Function to update the PID output.
+     * @param goal Final goal state for the controller.
+     * @param current Current state of the controller.
+     * @return Returns the Pid output
+     */
+    double Update(double goal, double current)
```

Comment:
Maybe provide the option to pass the time step in for the update computation.

Reply:
When I try to pass the time_step variable, I am also trying to set the default to class variable time_step_. However, it is not a static variable so it is allowing me to set this. Then if I make it static then I can't initialise it like [here](https://drake.mit.edu/styleguide/cppguide.html#Function_Names). I would have to set it in the constructor.

Reply:
No idea what you're asking here.  That link does not seem to be what you meant.

Reply:
My bad, I didn't double check the link I attached. 
Here is what I am taking about, the below picture is what you asked me to use for formatting. However, when I do this I get errors.
![old_implementation](https://user-images.githubusercontent.com/50416143/195906317-c8257a6a-9467-4434-aaf9-d1a691bce2c9.png)

So implemented it like this.
![code_implementation_current](https://user-images.githubusercontent.com/50416143/195906291-b87faa6d-1a9e-46f1-b6c8-108b99d1cb77.png)

Reply:
This does not seem to be a formatting question, but rather an issue with your implementation.  I can't say for sure without a full code snippet.  But the initializer list (the initial approach) is the way that you should be doing this, as discussed above.

Reply:
Okay.